### PR TITLE
Feat: Fetch fiat price

### PR DESCRIPTION
**Pull request by tmrdlt**
- Implemented `CoinmarketcapClient` for fetching fiat price on log basis including throtteling
- Introduced `HttpService`
- Bonus: Workaround for `int-as-string` param not being present in paginated calls

### CODE REVIEW

File: src/clients/mdw-http-client.service.ts [hunk 1]
```diff
@@ -81,23 +84,17 @@ export class MdwHttpClientService {
...
 
   // Fetches pages from middleware until the page contains at least one entry that meets the condition
   private async getPagesUntilCondition<T>(
     condition: (data: T) => boolean,
     next: string,
   ): Promise<T[]> {
-    const result = await this.get<MdwPaginatedResponse<T>>(next);
+    const result = await this.get<MdwPaginatedResponse<T>>(
+      next + `&int-as-string=${this.INT_AS_STRING}`,
+    );
```
Comment (kenodressel) [inline src/clients/mdw-http-client.service.ts:97]: minor notes: should have been its own PR

File: src/clients/coinmarketcap-client.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      );
+
+      await service.getHistoricalPriceDataThrottled(1704203614123);
+
+      expect(mockHttpService.get).toHaveBeenCalledWith(
+        'https://pro-api.coinmarketcap.com/v3/cryptocurrency/quotes/historical?id=1700&interval=24h&count=1&time_end=2024-01-02T13:50:00.000Z',
+        expect.anything(),
+      );
+    });
+
```
Comment (kenodressel) [inline src/clients/coinmarketcap-client.service.spec.ts:52]: should we test what happens if we run in the rate limit error?

File: src/clients/coinmarketcap-client.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      // Mock
+      mockHttpService.get.mockResolvedValue({});
+
+      // Call function
+      service.getHistoricalPriceDataThrottled(1704203935123);
+      service.getHistoricalPriceDataThrottled(1704203935123);
+      service.getHistoricalPriceDataThrottled(1704203935123);
+
+      await new Promise((res) => setTimeout(res, 200));
```
Comment (kenodressel) [inline src/clients/coinmarketcap-client.service.spec.ts:67]: seems a bit flaky, could we await the first two calls and then for the third call use the timeout?

**Review by kenodressel — APPROVED** [review]

File: src/clients/coinmarketcap-client.service.spec.ts (hunk 2 above)
Comment (tmrdlt) [reply src/clients/coinmarketcap-client.service.spec.ts:52]: Sounds good, I'd add it to the importer spec however, as in the CoinmarketcapClieht nothing much happens besides returning the error (we are using a mocked version of the API).

**Pull request merged**